h/api/storage.py

```diff
@@ -189,8 +189,12 @@ def delete_annotation(request, id):
     :param id: the annotation id
     :type id: str
     """
-    annotation = models.elastic.Annotation.fetch(id)
-    annotation.delete()
+    if _postgres_enabled(request):
+        annotation = fetch_annotation(request, id, _postgres=True)
+        request.db.delete(annotation)
+
+    legacy_annotation = fetch_annotation(request, id, _postgres=False)
```

Comment:
Note that I changed the legacy behaviour to call `fetch_annotation()` instead of calling `models.elastic.Annotation.fetch(id)`. I don't think this makes any difference, it's just more consistent and less duplication. Passing `_postgres=False` _is_ necessary here because the postgres feature flag might be true at this point but we want to fetch from Es not Pg.